torchgan/trainer/trainer.py

```diff
@@ -68,49 +59,55 @@ class Trainer(object):
 
     Example:
         >>> dcgan = Trainer(
-                    {"generator": {"name": DCGANGenerator, "args": {"out_channels": 1, "step_channels": 16}},
-                     "discriminator": {"name": DCGANDiscriminator, "args": {"in_channels": 1, "step_channels": 16}}},
-                    {"optimizer_generator": {"name": Adam, "args": {"lr": 0.0002, "betas": (0.5, 0.999)}},
-                     "optimizer_discriminator": {"name": Adam, "args": {"lr": 0.0002, "betas": (0.5, 0.999)}}},
+                    {"generator": {"name": DCGANGenerator, "args": {"out_channels": 1, "step_channels":
+                                   16}, "optimizer": {"name": Adam, "args": {"lr": 0.0002,
+                                   "betas": (0.5, 0.999)}}},
+                     "discriminator": {"name": DCGANDiscriminator, "args": {"in_channels": 1,
+                                       "step_channels": 16}, "optimizer": {"var": "opt_discriminator",
+                                       "name": Adam, "args": {"lr": 0.0002, "betas": (0.5, 0.999)}}}},
                     [MinimaxGeneratorLoss(), MinimaxDiscriminatorLoss()],
                     sample_size=64, epochs=20)
     """
-    def __init__(self, models, optimizers, losses_list, metrics_list=None, schedulers=None,
-                 device=torch.device("cuda:0"), ncritic=None, batch_size=128, epochs=5,
-                 sample_size=8, checkpoints="./model/gan", retain_checkpoints=5, recon="./images",
-                 log_dir=None, test_noise=None, nrow=8, **kwargs):
+    def __init__(self, models, losses_list, metrics_list=None, device=torch.device("cuda:0"),
+                 ncritic=None, batch_size=128, epochs=5, sample_size=8, checkpoints="./model/gan",
+                 retain_checkpoints=5, recon="./images", log_dir=None, test_noise=None, nrow=8, **kwargs):
         self.device = device
         self.model_names = []
-        for key, val in models.items():
+        self.optimizer_names = []
+        self.schedulers = []
+        for key, model in models.items():
             self.model_names.append(key)
-            if "args" in val:
-                setattr(self, key, (val["name"](**val["args"])).to(self.device))
+            if "args" in model:
+                setattr(self, key, (model["name"](**model["args"])).to(self.device))
             else:
-                setattr(self, key, (val["name"]()).to(self.device))
-        self.optimizer_names = []
-        for key, val in optimizers.items():
-            self.optimizer_names.append(key)
-            model = getattr(self, key.split("_", 1)[1])
-            if "args" in val:
-                setattr(self, key, val["name"](model.parameters(), **val["args"]))
+                setattr(self, key, (model["name"]()).to(self.device))
+            opt = model["optimizer"]
+            opt_name = "optimizer_{}".format(key)
+            if "var" in opt:
```

Comment:
I think its more natural to use "name" than "var" for naming the optimiser on the user end